### PR TITLE
Add selectable popup color palettes (#8)

### LINKED ISSUE #8: [Feature]: Add selectable popup color palettes
**Size:** large

## Status Quo

Walkthrough popups currently render with one purple-family color scheme. The popup is created through `ShowWalkthroughItemsToolset` and `WalkthroughOrchestrator`, hosted by `WalkthroughPopupSurface`, and drawn with Compose/Jewel content in `WalkthroughPopupContent.kt` and `WalkthroughPopupWidgets.kt`.

There is no plugin settings page today, so users cannot change the popup appearance. The relevant colors are defined in `WalkthroughColors.kt`, but the animated background, glow, border, scrollbar, navigation controls, and Swing connector still assume a single scheme. The popup animation was recently throttled through `ANIMATION_FRAME_INTERVAL_MS = 67L`, `TimeSource.Monotonic`, and `reverseLinearShift`; this feature should reuse that animation loop unchanged.

## Objectives

Give users a simple way to choose the popup color scheme without editing plugin code. The plugin should ship seven predefined palettes, keep the current purple scheme as the default, and persist the selected palette between IDE sessions. The settings UI should make the choices visual by showing each palette as a color swatch, not only as a text label.

## User Flow

**Trigger:** The user opens `Settings | Tools | Walkthrough`.

1. The settings page shows seven palette options: Purple, Green, Blue, Red, Orange, Teal, and Pink.
2. Each option includes a labelled gradient strip showing the palette colors.
3. The user selects a palette and applies the settings.
4. The next walkthrough popup reads the saved palette and renders the popup surface, controls, and connector with that scheme.

**Result:** Walkthrough popups use the selected palette, while Purple remains the default and preserves the current visual appearance.

## Scope

**In scope:**

- Add a dedicated `Palette` / `Palettes` model with seven predefined presets and stable persisted IDs.
- Preserve the current purple palette exactly as the default.
- Add application-level persisted settings for the selected palette ID.
- Add a `Settings | Tools | Walkthrough` configurable with labelled visual swatches.
- Refactor popup rendering so `WalkthroughPopupContent`, `WalkthroughPopupWidgets`, and `WalkthroughPopupSurface` consume a palette instead of assuming the current purple colors.
- Keep the popup animation cadence and geometry helpers unchanged.
- Add unit tests for preset content, unique IDs, lookup by ID, and unknown-ID fallback.

**Out of scope:**

- User-created custom palettes.
- Per-project palette selection.
- Per-project walkthrough history.
- Marketplace release/version bump.
- Changing popup animation timing, placement, or connector geometry.
- Live recoloring of an already open popup.

### Affected Modules

- `src/main/kotlin/com/forketyfork/walkthrough/WalkthroughPalette.kt`: add the palette value type and seven presets.
- `src/main/kotlin/com/forketyfork/walkthrough/WalkthroughColors.kt`: keep shared/static colors or move palette-specific colors into the new palette model.
- `src/main/kotlin/com/forketyfork/walkthrough/WalkthroughPopupContent.kt`: read background, glow, border, scrollbar, and meta colors from the selected palette.
- `src/main/kotlin/com/forketyfork/walkthrough/WalkthroughPopupWidgets.kt`: use palette-aware badge and navigation colors where needed.
- `src/main/kotlin/com/forketyfork/walkthrough/WalkthroughPopupSurface.kt`: use palette-aware connector stroke and arrow colors.
- `src/main/kotlin/com/forketyfork/walkthrough/WalkthroughSettings.kt`: add persisted application-level settings for the selected palette ID.
- `src/main/kotlin/com/forketyfork/walkthrough/WalkthroughSettingsConfigurable.kt`: add the settings UI with visual swatches.
- `src/main/resources/META-INF/plugin.xml`: register the settings service and configurable.
- `src/test/kotlin/com/forketyfork/walkthrough/WalkthroughPaletteTest.kt`: cover presets and lookup behavior.

### Tasks

Each task is independently testable. Ordered by implementation sequence.

1. [ ] Add the palette model and seven presets, including an exact current-purple default — `WalkthroughPalette.kt`
2. [ ] Add palette lookup by stable ID with default fallback — `WalkthroughPalette.kt`
3. [ ] Add application-level persisted settings for the selected palette ID — `WalkthroughSettings.kt`
4. [ ] Register the settings service and configurable — `plugin.xml`
5. [ ] Add a Jewel-based settings panel under `Settings | Tools | Walkthrough` with labelled gradient swatches — `WalkthroughSettingsConfigurable.kt`
6. [ ] Thread the selected palette into popup creation and rendering without changing `ANIMATION_FRAME_INTERVAL_MS`, `TimeSource.Monotonic`, or `reverseLinearShift` — `WalkthroughOrchestrator.kt`, `WalkthroughPopupContent.kt`, `WalkthroughPopupWidgets.kt`, `WalkthroughPopupSurface.kt`
7. [ ] Write tests: seven presets exist, IDs are unique and stable, Purple is the default, unknown IDs fall back to Purple, and current purple values are preserved — `WalkthroughPaletteTest.kt`
8. [ ] Update docs/PRD.md (if present)
9. [ ] Update docs/TRACEABILITY.md (if present)
10. [ ] Update docs/ARCHITECTURE.md (if architecture changed and file present)

### New Dependencies

None. Use IntelliJ Platform settings APIs and the existing Compose/Jewel UI stack.

## Acceptance Criteria

- [ ] All tasks completed
- [ ] Users can pick one of seven predefined palettes from `Settings | Tools | Walkthrough`
- [ ] Each palette option is shown with a labelled color swatch
- [ ] The selected palette persists across IDE restarts
- [ ] Purple is the default and preserves the current popup appearance
- [ ] Unknown or missing palette IDs fall back to Purple
- [ ] The popup animation throttle remains at `ANIMATION_FRAME_INTERVAL_MS = 67L` and still uses `TimeSource.Monotonic` with `reverseLinearShift`
- [ ] Tests cover happy path and documented edge cases
- [ ] docs/PRD.md updated (if present)
- [ ] docs/TRACEABILITY.md updated (if present)
- [ ] Pre-commit hooks pass

### PR DESCRIPTION
**Pull request by forketyfork**
## Solution

This adds a Walkthrough settings page with seven built-in popup palettes: Purple, Green, Blue, Red, Orange, Teal, and Pink. Purple stays the default, and its colors match the previous hard-coded values.

The selected palette is stored as application-level state using a stable palette ID. When a walkthrough popup opens, the orchestrator reads that palette once and passes it through the Compose content, navigation widgets, and Swing connector painting. This keeps the agreed behavior: palette changes apply to the next popup, while an already open popup is left alone.

The popup animation code is intentionally unchanged. `ANIMATION_FRAME_INTERVAL_MS`, `TimeSource.Monotonic`, `reverseLinearShift`, and `WalkthroughPopupGeometry.kt` are not touched.

Closes #8

## Test plan

- [x] Open `Settings | Tools | Walkthrough` and confirm all seven palettes appear with labelled gradient swatches.
- [x] Select a non-purple palette, apply settings, then open a walkthrough popup and confirm the popup uses that palette.
- [x] Select Purple again and confirm the popup returns to the original purple look.

### CODE REVIEW

File: src/main/kotlin/com/forketyfork/walkthrough/WalkthroughSettings.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.forketyfork.walkthrough
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+
+@State(
+    name = "WalkthroughSettings",
+    storages = [Storage("walkthrough.xml")]
```
[the remark this answers is not in the document]
Comment (forketyfork) [reply src/main/kotlin/com/forketyfork/walkthrough/WalkthroughSettings.kt]: Renamed the persistent state and storage file to plugin-qualified values in `881d3be`.

File: src/main/kotlin/com/forketyfork/walkthrough/WalkthroughSettingsConfigurable.kt [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        selectedPaletteId = WalkthroughSettings.getInstance().selectedPaletteId
+    }
+
+    override fun reset() {
+        selectedPaletteId = WalkthroughSettings.getInstance().selectedPaletteId
+    }
+
+    override fun disposeUIResources() {
+        component = null
+    }
```
[the remark this answers is not in the document]
Comment (forketyfork) [reply src/main/kotlin/com/forketyfork/walkthrough/WalkthroughSettingsConfigurable.kt:114]: Removed the unused component field and the redundant `disposeUIResources()` override in `881d3be`.
